src/clr_bundle_rm.c

```diff
@@ -121,56 +142,150 @@ static bool parse_options(int argc, char **argv)
 				printf("Invalid --format argument\n\n");
 				goto err;
 			}
+			string_or_die(&curopts.format_string, "%s", optarg);
 			break;
 		case 'S':
 			if (!optarg || !set_state_dir(optarg)) {
 				printf("Invalid --statedir argument\n\n");
 				goto err;
 			}
+			string_or_die(&curopts.state_dir, "%s", optarg);
 			break;
 		case 'x':
 			force = true;
+			curopts.force = true;
 			break;
 		case 'n':
 			sigcheck = false;
+			curopts.sigcheck = false;
 			break;
 		case 'C':
 			if (!optarg) {
 				printf("Invalid --certpath argument\n\n");
 				goto err;
 			}
 			set_cert_path(optarg);
+			string_or_die(&curopts.cert_path, "%s", optarg);
 			break;
 		default:
 			printf("error: unrecognized option\n\n");
 			goto err;
 		}
 	}
 
-	if (argc != optind + 1) {
-		printf("error: invalid arguments\n\n");
+	if (argc <= optind) {
+		printf("error: missing bundle(s) to be removed\n\n");
 		goto err;
 	}
-	string_or_die(&bundle_name, "%s", argv[optind]);
+
+	bundles = argv + optind;
 
 	return true;
 err:
 	print_help(argv[0]);
 	return false;
 }
 
+static void reload_parsed_opts(void)
+{
+	if (curopts.path_prefix) {
+		set_path_prefix(curopts.path_prefix);
+	}
+
+	if (curopts.version_url) {
+		set_version_url(curopts.version_url);
+	}
+
+	if (curopts.content_url) {
+		set_content_url(curopts.content_url);
+	}
+
+	if (curopts.format_string) {
+		set_format_string(curopts.format_string);
+	}
+
+	if (curopts.state_dir) {
+		set_state_dir(curopts.state_dir);
+	}
+
+	if (curopts.cert_path) {
+		set_cert_path(curopts.cert_path);
+	}
+
+	force = curopts.force;
+	sigcheck = curopts.sigcheck;
+
+}
+
+void static free_saved_opts(void)
+{
+	if (curopts.path_prefix != NULL) {
+		free(curopts.path_prefix);
+	}
+
+	if (curopts.version_url != NULL) {
+		free(curopts.version_url);
+	}
+
+	if (curopts.content_url != NULL) {
+		free(curopts.content_url);
+	}
+
+	if (curopts.format_string != NULL) {
+		free(curopts.format_string);
+	}
+
+	if (curopts.state_dir != NULL) {
+		free(curopts.state_dir);
+	}
+
+	if (curopts.cert_path != NULL) {
+		free(curopts.cert_path);
+	}
+
+
+}
+
 int bundle_remove_main(int argc, char **argv)
 {
-	int ret;
+	int ret = 0;
+	int total = 0;
+	int bad = 0;
 
 	copyright_header("bundle remover");
 
 	if (!parse_options(argc, argv)) {
 		return EINVALID_OPTION;
 	}
 
-	ret = remove_bundle(bundle_name);
-	free(bundle_name);
+	for (; *bundles; ++bundles, total++) {
+		printf("Removing bundle: %s\n", *bundles);
+
+		if (remove_bundle(*bundles) != 0) {
```

Comment:
I agree and in fact this is just a dumb wrapper around bundle-remove to align consistency of use along with bundle-add, I know there is an optimal solution and doesn't even need any wrappers, however, I decided to go this path for two reasons:
1) the less the core of remove_bundle() core function changes the more the chance to no break anything else (overall with includes stuff and the so), then having it working on multiple removes we can add more tests over that functionality, the test coverages will ensure that in the future "optimized" function does not break anything.
2) Wanted to get this as soon as possible (without a broken swupd-remove of course) for consistency and usability, then we can go from this to implement optimal solution.

Having said that, I don't want to try optimizing anymore this wrapper, what I'd expect is to have it merged, rise test coverage and start developing the optimized solution that would for sure change the core heart of remove_bundle().